### PR TITLE
Remove webpack-dev-server

### PR DESCRIPTION
**Pull request by nkzawa**
Fixes https://github.com/zeit/next.js/issues/236, https://github.com/zeit/next.js/issues/102

By using [`webpack-hot-middleware`](https://github.com/glenjamin/webpack-hot-middleware), we can use HMR from our server instead of running a dedicated server (webpack-dev-server).

### CODE REVIEW

File: server/build/webpack.js [hunk 1]
```diff
@@ -144,7 +146,7 @@ export default async function createCompiler (dir, { hotReload = false, dev = fa
       path: join(dir, '.next'),
       filename: '[name]',
       libraryTarget: 'commonjs2',
-      publicPath: hotReload ? 'http://localhost:3030/' : null
+      publicPath: hotReload ? '/_webpack' : null
```
Comment (cusxio) [inline server/build/webpack.js]: shouldn't it be `/_webpack/` ?

Comment (nkzawa) [inline server/build/webpack.js]: yeah, fixed, thanks!

**Comment by jonaswindey** [comment]
Trying to run a locally built version with `node dist/bin/next`

getting error
```
ERROR in multi bundles/pages/_error.js
Module not found: Error: Cannot resolve module 'next/dist/client/webpack-hot-middleware-client' in /Users/jonas/Development/github/next.js
 @ multi bundles/pages/_error.js
```

**Comment by jonaswindey** [comment]
Also, when adding a new page (test.js) in /dist/pages

```
import React from 'react'

export default () => <div>hi</div>
```

I get the following error:

```
ERROR in multi bundles/pages/test.js
Module not found: Error: Cannot resolve module 'next/dist/client/webpack-hot-middleware-client' in /Users/jonas/Development/github/next.js/dist
 @ multi bundles/pages/test.js
> Ready on http://localhost:3000
/Users/jonas/Development/github/next.js/dist/server/build/plugins/watch-pages-plugin.js:53
          return _this.prevFileDependencies.indexOf(f) < 0;
                                           ^

TypeError: Cannot read property 'indexOf' of null
    at /Users/jonas/Development/github/next.js/dist/server/build/plugins/watch-pages-plugin.js:53:44
    at Array.filter (native)
    at Compiler.<anonymous> (/Users/jonas/Development/github/next.js/dist/server/build/plugins/watch-pages-plugin.js:52:73)
    at Compiler.applyPluginsAsync (/Users/jonas/Development/github/next.js/node_modules/tapable/lib/Tapable.js:71:13)
    at Watching._go (/Users/jonas/Development/github/next.js/node_modules/webpack/lib/Compiler.js:44:16)
    at Watching.invalidate (/Users/jonas/Development/github/next.js/node_modules/webpack/lib/Compiler.js:108:8)
    at Watching.<anonymous> (/Users/jonas/Development/github/next.js/node_modules/webpack/lib/Compiler.js:93:8)
    at /Users/jonas/Development/github/next.js/dist/server/build/plugins/watch-remove-event-plugin.js:45:22
    at NodeWatchFileSystem.<anonymous> (/Users/jonas/Development/github/next.js/node_modules/webpack/lib/node/NodeWatchFileSystem.js:43:3)
    at EventEmitter.g (events.js:286:16)
```

**Comment by jonaswindey** [comment]
@nkzawa

**Comment by nkzawa** [comment]
@jonaswindey hmm, I can't reproduce. I did clean install (rm -rf node_modules && npm i). How did you install `next` locally ?

**Comment by jonaswindey** [comment]
```
mkdir tmp
git clone https://github.com/zeit/next.js.git tmp && cd tmp
npm i
node dist/bin/next
```

```
ERROR in multi bundles/pages/_error.js
Module not found: Error: Cannot resolve module 'next/dist/client/webpack-hot-middleware-client' in /Users/jonas/Development/sandbox/tmp
 @ multi bundles/pages/_error.js
> Ready on http://localhost:3000
```

**Comment by nkzawa** [comment]
@jonaswindey it looks you executed `dist/bin/next` on the root directory of `next.js` ?
You have to do it on the directory of your app, or pass a directory to the command argument like `./dist/bin/next ../../my-next-app`.

**Comment by jonaswindey** [comment]
@nkzawa if I do this on the latest commit from yesterday that command still works, and not anymore.

this was the command I use to run next locally and this allowed me to test pages when creating them under /dist/pages

how do you run 'next' locally when you want to test your own examples?

**Comment by nkzawa** [comment]
@jonaswindey these are not supported even if it works before by accident. I recommend to use `npm link` which allows to install local modules.

**Pull request merged**